js_modules/dagster-ui/packages/ui-core/src/schedules/ScheduleSwitch.tsx

```diff
@@ -57,12 +56,16 @@ export const ScheduleSwitch = (props: Props) => {
     StartThisScheduleMutationVariables
   >(START_SCHEDULE_MUTATION, {
     onCompleted: displayScheduleMutationErrors,
+    refetchQueries: [{variables: {id: schedule.id}, query: SCHEDULE_STATE_QUERY}],
+    awaitRefetchQueries: true,
   });
   const [stopSchedule, {loading: toggleOffInFlight}] = useMutation<
     StopScheduleMutation,
     StopScheduleMutationVariables
   >(STOP_SCHEDULE_MUTATION, {
     onCompleted: displayScheduleMutationErrors,
+    refetchQueries: [{variables: {id: schedule.id}, query: SCHEDULE_STATE_QUERY}],
+    awaitRefetchQueries: true,
```

Comment:
This is to ensure the switch doesn't temporarily show up as "off" while we refetch the query. The refetch is necessary because if ScheduleStateQuery previously received an `InstigationNotFoundError` then the mutation doesn't end up updating the cache here. This does mean that updates from elsewhere in the app wouldn't propagate here but this is the only component that calls this mutation so I don't think we need a useFragment watching approach here.

Reply:
Oh wow I didn't realize this was an available option, this seems like an ideal use case for it 👍